### PR TITLE
Add prefix_length member variable to Ipv4Prefix

### PR DESCRIPTION
**Pull request by gampel**

### CODE REVIEW

**Comment by changlan** [comment]
Thanks for the PR. I am not clear about the purpose of this patch. Isn't `prefix_length` implied by the value of `mask` already?

**Comment by sangjinhan** [comment]
Yeah, perhaps you can add a method using __builtin_ffs() to retrieve the prefix length.

**Comment by gampel** [comment]
Will do, I will add a method using the __builtin

File: core/utils/ip.cc [hunk 1]
```diff
@@ -52,6 +52,7 @@ Ipv4Prefix::Ipv4Prefix(const std::string &prefix) {
   } else {
     mask = be32_t(~((1 << (32 - len)) - 1));
   }
+
```
Comment (changlan) [inline core/utils/ip.cc]: redundant linebreak?

File: core/utils/ip.h [hunk 2]
```diff
@@ -67,6 +67,8 @@ struct Ipv4Prefix {
 
   // Returns true if ip is within the range of Ipv4Prefix
   bool Match(const be32_t &ip) const { return (addr & mask) == (ip & mask); }
+  // Returns the prefix length
+  uint32_t prefixLength() const {return ((32 - __builtin_clz(mask.raw_value())));}
```
Comment (changlan) [inline core/utils/ip.h]: Perhaps rename the function to `prefix_length` (accessors may be named like variables as per Google C++ Style Guide)

Comment (gampel) [inline core/utils/ip.h]: Done

File: core/utils/ip.cc (hunk 1 above)
Comment (gampel) [inline core/utils/ip.cc]: will be removed

**Comment by changlan** [comment]
Merged, thanks!
image

**Pull request merged**